example/standalone-user-provided-creds/main.tf

```
@@ -1,30 +1,41 @@
 module "minio" {
-  source = "github.com/fredrikhgrelland/terraform-nomad-minio.git?ref=0.2.0"
+  source = "github.com/fredrikhgrelland/terraform-nomad-minio.git?ref=0.3.0"
 
   # nomad
   nomad_datacenters = ["dc1"]
   nomad_namespace   = "default"
   nomad_host_volume = "persistence-minio"
 
   # minio
-  service_name                    = "minio"
-  host                            = "127.0.0.1"
-  port                            = 9000
-  container_image                 = "minio/minio:latest" # todo: avoid using tag latest in future releases
-  access_key                      = "minio"
-  secret_key                      = "minio123"
+  service_name    = "minio"
+  host            = "127.0.0.1"
+  port            = 9000
+  container_image = "minio/minio:latest" # todo: avoid using tag latest in future releases
+  # user provided  credentials
+  vault_secret = {
+    use_vault_provider   = false,
+    vault_kv_policy_name = "",
+    vault_kv_path          = "",
+    vault_kv_access_key    = "",
+    vault_kv_secret_key    = ""
+  }
+  access_key = "minio"
+  secret_key = "minio123"
```

Make a clear separation between box's minio, and the minio module
```suggestion
  box_minio_access_key = "minio"
  box_minio_secret_key = "minio123"
```

---

issue is created https://github.com/fredrikhgrelland/terraform-nomad-minio/issues/87

---

It is not necessary, because box's minio creds are hardcoded in `hive.hcl`, we do not have variables for that ,
```hcl
      artifact {
        source = "s3::http://127.0.0.1:9000/dev/tmp/hive_local.tar"
        options {
          aws_access_key_id     = "minioadmin"
          aws_access_key_secret = "minioadmin"
        }
      }
```

---

oh, well, just remove it completely then!

---

We need a way how to test the newly build image, without pulling from the docker registry.

This functionality uses only with box, but we still need it, if we want to test changes in docker image, before releasing docker image and pushing it to the registry.

---

Yeah, I didn't mean remove the artifact stanza, I meant the variables set in the .tf file

---

I have provided comments about it, by ur request. I hoped it could be better explained that way https://github.com/fredrikhgrelland/terraform-nomad-hive/blob/issue_44_creds_management/example/standalone-vault-provided-creds/main.tf#L15-L25